### DISCUSSION TITLE
Copier update not working properly under a conditional directory

### DISCUSSION THREAD

**Discussion by guillemrbaiges** [copier-org/copier#885] [Q&A]
The context is as follows

Given:
- Copier version is v6
- the Copier project I am trying to update contains a different set of questions than the latest version of its original Copier template
- the directory having the issue has the following name syntax: `{% if copier_bool_variable %}directory_name{% endif %}`
- the directory contains 4 files in latest version, 3 files in the project we're trying to update
- some of these files have updates that should be reflected in the project after the `copier update` call

Expected:
- Project properly updated after running `copier update --force --data ...` (as per your [documentation](https://copier.readthedocs.io/en/stable/updating/#never-change-the-answers-file-manually))

What happens:
- (In the mentioned directory) Some files get deleted, the updates in the files are not correct (results in 2 files instead of the original 3 or the expected 4)

Weirdly, if I ignore all recommendations and warnings and I manually add the missing questions in the `copier-answers` file, then this behaviour does not happen and I get the 4 files I was expecting. I don't remember if this strategy ended up working, as I recall at some points the expected changes appeared mid-update and then disappeared (I can't replicate this right now as this is an ongoing collaboration with one of our clients), but I'm 100% sure that the weird behaviour described above (files wrongly removed) does not happen when I manually modify the `copier-answers` file to add the new questions, instead of passing them through the `--data` command.

Let me know if you need more information on the issue, I've mentioned my two main suspicions (different questions to the ones in the latest version + update under a directory whose name uses Copier's conditional syntax) to see if this rings a bell for you. Thanks!!

**Comment by yajo** [comment]
I'm not sure I understand fully the case but in case it's a bug in copier you're probably hitting some corner case that hasn't been tested so far. If there was a way to get reproduction steps, that'd help a lot in proper diagnose.
